File: trials/queue_array.c
//Author: Sanskriti Gupta
//Date: 16 April 2022
//Aim: To implement queue using array
#include<stdio.h>
#include<stdlib.h>
#define capacity 100
int queue[capacity];
int front=-1,rear=-1;
void enqueue(int item)
{
    if(rear==capacity)
    {
        printf("\nOVERFLOW");
    }
    else if(front==-1)
    {
        front=0;
        rear++;
        queue[rear]=item;
    }
    else
    {
        rear++;
        queue[rear]=item;
    }
}
int dequeue()
{
    if(front==-1)
    {
        printf("\nUNDERFLOW");
    }
    else if(front>=0)
    {
        int i=queue[front];
        front++;
        return i;
    }
}
void display()
{
    for(int i=front;i<=rear;i++)
    {
        printf("%d ",queue[i]);
    }
}
void main()
{
    int choice, item;
    while(1)
    {
        printf("\n1. Enqueue");
        printf("\n2. Dequeue");
        printf("\n3. Display");
        printf("\n4. Exit\n");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1:
                printf("\nEnter the element: ");
                scanf("%d",&item);
                enqueue(item);
                break;
            case 2:
                item=dequeue();
                printf("\nDeleted Element: %d",item);
                break;
            case 3:
                display();
                break;
            case 4:
                exit(0);
            default:
                printf("\nWrong Choice");
        }
    }
}

File: java/queue_array.java

//YE WALA GALAT HAI

import java.util.Scanner;
import java.lang.System;
public class queue_array
{
    public static int n;
    public static int front = -1;
    public static int rear = -1;
    public static void main(String[] args)
    {
        Scanner var = new Scanner(System.in);
        try
        {
            System.out.print("\nEnter the limit of queue: ");
            n = var.nextInt();
            int[] arr = new int[n];
            while(true)
            {
                System.out.print("\n1. Enqueue");
                System.out.print("\n2. Dequeue");
                System.out.print("\n3. Display");
                System.out.print("\n4. Exit\n");
                int choice = var.nextInt();
                switch(choice)
                {
                    case 1:
                        System.out.print("\nEnter an element: ");
                        int item = var.nextInt();
                        enqueue(arr, item);
                        break;
                    case 2:
                        int del = dequeue(arr);
                        System.out.print("\nDeleted Item: " + del);
                        break;
                    case 3:
                        display(arr);
                        break;
                    case 4:
                        System.exit(0);
                    default:
                        System.out.print("\nNo such choice, try again !!!");
                }
            }
        }
        finally
        {
            var.close();
        }
    }
    public static void enqueue(int[] arr, int item)
    {
        if(rear == n)
        {
            System.out.print("\nOVERFLOW !!!");
        }
        else if(front == -1 && rear == -1)
        {
            front = 0;
            rear++;
            arr[rear] = item;
        }
        else
        {
            rear++;
            arr[rear] = item;
        }
    }
    public static int dequeue(int[] arr)
    {
        if(front == -1)
        {
            System.out.print("\nUNDERFLOW !!!");
            return 0;
        }
        else
        {
            int i = arr[front];
            front++;
            return i;
        }
    }
    public static void display(int[] arr)
    {
        if(front == -1)
        {
            System.out.print("EMPTY QUEUE !!!");
        }
        int i;
        for(i = front; i <= rear; i++)
        {
            System.out.print(arr[i] + " ");
        }
    }
}